src/raft.rs

```diff
@@ -789,7 +795,14 @@ impl<T: Storage> RaftCore<T> {
                 return false;
             }
         } else {
-            let ents = self.raft_log.entries(pr.next_idx, self.max_msg_size);
+            let ents = self.raft_log.entries(
+                pr.next_idx,
+                self.max_msg_size,
+                GetEntriesContext(GetEntriesFor::SendAppend {
+                    to,
+                    aggressively: !allow_empty,
```

Comment:
Why let `allow_empty` decide whether send append log aggressively?

Reply:
`allow_empty` only set to `false` when sending log aggressively, you can check the comment of `maybe_send_append`